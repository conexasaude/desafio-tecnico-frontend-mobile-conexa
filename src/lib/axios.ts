import axios from 'axios';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { storageKeys } from '@/lib/constants/storageKeys';
import { BASE_URL } from '@env';
import { User } from '@/models/user';

export const api = axios.create({ baseURL: BASE_URL });

api.interceptors.request.use(
  async config => {
    const data = await AsyncStorage.getItem(storageKeys.user);
    const user = data != null ? (JSON.parse(data) as User) : null;

    if (user) {
      config.headers.Authorization = `Bearer ${user.token}`;
    }

    return config;
  },
  error => {
    return Promise.reject(error);
  },
);
