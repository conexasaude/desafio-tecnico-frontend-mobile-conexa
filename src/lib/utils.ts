export interface ThunkCallback {
  onSuccess?: () => void;
  onError?: (message: string) => void;
}

export function formatDateToISO(dateString: string): string {
  const parts = dateString.split('/');

  if (parts.length !== 3) {
    return dateString;
  }

  const day = parts[1].padStart(2, '0');
  const month = parts[0].padStart(2, '0');
  const year = parts[2];

  return `${year}-${day}-${month}`;
}

export function formatDateToLocale(dateString: string, device: string): string {
  const parts = dateString.split('/');
  const isIOS = device === 'IOS';

  if (parts.length !== 3) {
    return dateString;
  }

  const day = isIOS ? parts[0].padStart(2, '0') : parts[1].padStart(2, '0');
  const month = isIOS ? parts[1].padStart(2, '0') : parts[0].padStart(2, '0');
  const year = parts[2];

  return `${day}/${month}/${year}`;
}

export function formatTime(time: string): string {
  const parts = time.split(':');
  const hours = parseInt(parts[0], 10);
  const minutes = parseInt(parts[1], 10);

  let formateMinutes = minutes.toString().padStart(2, '0');
  let formattedHours = hours.toString().padStart(2, '0');

  if (time.includes('PM') && hours !== 12) {
    formattedHours = (hours + 12).toString().padStart(2, '0');
  } else if (time.includes('AM') && hours === 12) {
    formattedHours = '00';
  }

  return `${formattedHours}:${formateMinutes}`;
}
