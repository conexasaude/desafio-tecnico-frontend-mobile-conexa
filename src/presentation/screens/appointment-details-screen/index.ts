export * from './appointment-details-screen';
