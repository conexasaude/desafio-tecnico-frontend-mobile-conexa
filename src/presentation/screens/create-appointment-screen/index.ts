export * from './create-appointment-screen';
