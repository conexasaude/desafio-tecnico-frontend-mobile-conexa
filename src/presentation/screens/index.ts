export * from './appointments-screen';
export * from './create-appointment-screen';
export * from './login-screen';
