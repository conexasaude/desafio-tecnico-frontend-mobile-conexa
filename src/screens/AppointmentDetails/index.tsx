import { View } from 'react-native';
import { NativeStackScreenProps } from "@react-navigation/native-stack";
import { FontAwesome } from '@expo/vector-icons';
import RootStackParamList from '../../types/rootStackParamList';
import formatsDate from '../../utils/formatsDate';
import { Container, Content, Field, FieldContainer, Header, FieldText, Label } from './styles';
import { useEffect } from 'react';

type Props = NativeStackScreenProps<RootStackParamList, 'AppointmentDetails'>;

export default function AppointmentDetails({ route, navigation }: Props) {
	const { appointment } = route.params;
	
	useEffect(() => {
		navigation.setOptions({ title: `Consulta ${appointment.id}` })
	}, [])

	return (
		<Container>
			<Header>
				<View>
					<FontAwesome name="user-circle-o" size={72} color="black" />
				</View>
				<FieldContainer>
					<Field>
						<Label>Paciente:</Label>
						<FieldText>{appointment.paciente}</FieldText>
					</Field>

					<Field>
						<Label>Data e hora: </Label>
						<FieldText>{`${formatsDate(appointment.dataConsulta, 'weekday-date-time')}`}</FieldText>
					</Field>
				</FieldContainer>
			</Header>
				
			<Content>
				<Field>
					<Label>Observação: </Label>
					<FieldText>{appointment.observacao}</FieldText>
				</Field>
			</Content>
		</Container>
    )
}