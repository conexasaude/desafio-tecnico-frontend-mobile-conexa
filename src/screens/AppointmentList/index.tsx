import { FlatList, View } from 'react-native';
import { NativeStackScreenProps } from "@react-navigation/native-stack";
import { useEffect, useState } from 'react';
import Spinner from 'react-native-loading-spinner-overlay';
import { getAppointments } from '../../services/appointment';
import { Appointment } from '../../types/appointment';
import RootStackParamList from '../../types/rootStackParamList';
import AppointmentCard from '../../components/AppointmentCard';
import SearchInput from '../../components/SearchInput';
import { Container } from './styles';

type Props = NativeStackScreenProps<RootStackParamList, 'AppointmentList'>;

export default function AppointmentList({ navigation }: Props) {
	const [appointments, setAppointments] = useState<Appointment[]>([])
	const [loading, setLoading] = useState(false)
	const [search, setSearch] = useState("")

	async function fetchAppointments() {
		setLoading(true)
		const response = await getAppointments()
		setAppointments(response)
		setLoading(false)
	}

	async function navigateToDetails(appointment: Appointment) {
		navigation.navigate('AppointmentDetails', {
			appointment
		})
	}

	function filterAppointments(appointments: Appointment[]) {
		if(search !== '') {
			return appointments.filter(appointment => 
				String(appointment.id).toLocaleLowerCase().includes(search.toLocaleLowerCase())  || appointment.paciente.toLocaleLowerCase().includes(search.toLocaleLowerCase())
			)
		} else {
			return appointments
		}
	}
    
	useEffect(() => {
		fetchAppointments()
	}, [])

	return (
		<>
			<Spinner
				visible={loading}
				textContent={'Carregando...'}
				textStyle={{color: '#FFF'}}
			/>
				
			<Container>
				<SearchInput value={search} onChangeText={setSearch} />

				<FlatList
					data={filterAppointments(appointments)}
					renderItem={({item}) => 
						<AppointmentCard
							id={String(item.id)}
							onPress={() => navigateToDetails(item)} 
							pacient={item.paciente}
							date={item.dataConsulta}
						/>
					}
					style={{ paddingHorizontal: 16 }}
				>
				</FlatList>
			</Container>
		</>
	)
}