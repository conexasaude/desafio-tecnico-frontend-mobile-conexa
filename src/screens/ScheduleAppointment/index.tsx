import React, { useCallback } from 'react';
import * as Styled from './styles';
import { Input } from '@/components/Input';
import { DatePicker } from '@/components/DatePicker';
import { Button } from '@/components/Button';
import appointmentSchema, {
  FormScheduleAppointment,
} from '@/validations/appointment';
import { useFormik } from 'formik';
import { toFormikValidationSchema } from 'zod-formik-adapter';
import { useReduxSelector } from '@/hooks/useReduxSelector';
import { useReduxDispatch } from '@/hooks/useReduxDispatch';
import { createAppointment } from '@/store/thunks/appointment.thunk';
import { useToast } from 'react-native-toast-notifications';
import { useNavigation } from '@react-navigation/native';
import { NavPropsLogged } from '@/routes/logged';

export function ScheduleAppointmentScreen() {
  const { show } = useToast();
  const { loading } = useReduxSelector(state => state.appointment);
  const { navigate } = useNavigation<NavPropsLogged>();
  const dispatch = useReduxDispatch();

  const initialValues = {
    patient: '',
    date: '',
    time: '',
    observation: '',
  };

  const onSubmit = useCallback(
    async (data: FormScheduleAppointment) => {
      dispatch(
        createAppointment({
          data,
          onError: show,
          onSuccess: () => navigate('Appointments'),
        }),
      );
    },
    [dispatch, navigate, show],
  );

  const { handleChange, handleSubmit, values, errors, submitCount } = useFormik(
    {
      initialValues,
      onSubmit,
      validationSchema: toFormikValidationSchema(appointmentSchema),
    },
  );

  return (
    <Styled.SafeArea>
      <Styled.Container>
        <Styled.Description>
          Preencha os campos abaixo para agendar uma consulta com o paciente
        </Styled.Description>

        <Styled.Form>
          <Input
            testID="input-patient"
            label="Nome do paciente"
            placeholder="Ex: João da silva"
            value={values.patient}
            onChangeText={handleChange('patient')}
            error={submitCount && errors.patient ? errors.patient : ''}
          />
          <Styled.Row>
            <DatePicker
              testID="date-picker-date"
              label="Data"
              mode="date"
              onChange={handleChange('date')}
              error={submitCount && errors.date ? errors.date : ''}
            />

            <DatePicker
              testID="date-picker-time"
              label="Hora"
              mode="time"
              onChange={handleChange('time')}
              error={submitCount && errors.time ? errors.time : ''}
            />
          </Styled.Row>

          <Input
            testID="input-observation"
            label="Observação"
            placeholder="..."
            textArea
            value={values.observation}
            onChangeText={handleChange('observation')}
            error={submitCount && errors.observation ? errors.observation : ''}
          />
        </Styled.Form>

        <Styled.Spacer />

        <Button
          title="Agendar"
          onPress={() => handleSubmit()}
          loading={loading}
        />
      </Styled.Container>
    </Styled.SafeArea>
  );
}
