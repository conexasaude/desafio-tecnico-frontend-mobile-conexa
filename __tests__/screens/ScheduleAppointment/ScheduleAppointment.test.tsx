import React from 'react';
import MockAdapter from 'axios-mock-adapter';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/lib/functions/Testing';

import { ScheduleAppointmentScreen } from '@/screens/ScheduleAppointment';
import { fireEvent, waitFor } from '@testing-library/react-native';
import { api } from '@/lib/axios';

const mockDispatch = jest.fn();
const mockNavigate = jest.fn();
const mockShowToast = jest.fn();

jest.mock('@/hooks/useReduxDispatch', () => ({
  useReduxDispatch: () => mockDispatch,
}));

jest.mock('@/hooks/useReduxSelector', () => ({
  useReduxSelector: () => ({ loading: false }),
}));

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ navigate: mockNavigate }),
}));

jest.mock('react-native-toast-notifications', () => ({
  useToast: () => ({ show: mockShowToast }),
}));

describe('ScheduleAppointment', () => {
  test('should render correctly', () => {
    renderWithThemeProvider(<ScheduleAppointmentScreen />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<ScheduleAppointmentScreen />);
  });

  test('should dispatch create appointment and navigate back to appointments', () => {
    const { getByText, getByTestId } = renderWithThemeProvider(
      <ScheduleAppointmentScreen />,
    );

    const inpuPatient = getByTestId('input-patient');
    const datePickerDate = getByTestId('date-picker-date');
    const datePickerTime = getByTestId('date-picker-time');
    const inputObservation = getByTestId('input-observation');

    const submitButton = getByText('Agendar');

    fireEvent.changeText(inpuPatient, 'Maria Brown');

    fireEvent.press(datePickerDate);
    const confirmDateButton = getByText('confirm');
    fireEvent.press(confirmDateButton);

    fireEvent.press(datePickerTime);
    const confirmTimeButton = getByText('confirm');
    fireEvent.press(confirmTimeButton);

    fireEvent.changeText(inputObservation, 'something');

    fireEvent.press(submitButton);

    waitFor(() => expect(mockDispatch).toHaveBeenCalled());
    waitFor(() => expect(mockNavigate).toHaveBeenCalled());
  });

  test('should call toast if failead to create appointment', () => {
    const mockAxios = new MockAdapter(api);
    const mockError = 'Request failed with status code 500';

    mockAxios.onPost('/api/consulta').reply(500, { error: mockError });

    const { getByText, getByTestId } = renderWithThemeProvider(
      <ScheduleAppointmentScreen />,
    );

    const inpuPatient = getByTestId('input-patient');
    const datePickerDate = getByTestId('date-picker-date');
    const datePickerTime = getByTestId('date-picker-time');
    const inputObservation = getByTestId('input-observation');

    const submitButton = getByText('Agendar');

    fireEvent.changeText(inpuPatient, 'Maria Brown');

    fireEvent.press(datePickerDate);
    const confirmDateButton = getByText('confirm');
    fireEvent.press(confirmDateButton);

    fireEvent.press(datePickerTime);
    const confirmTimeButton = getByText('confirm');
    fireEvent.press(confirmTimeButton);

    fireEvent.changeText(inputObservation, 'something');

    fireEvent.press(submitButton);

    waitFor(() => expect(mockShowToast).toHaveBeenCalled());
  });

  test('should show an error message if you don not fill in the fields', () => {
    const { getByText } = renderWithThemeProvider(
      <ScheduleAppointmentScreen />,
    );

    const submitButton = getByText('Agendar');

    fireEvent.press(submitButton);

    waitFor(() => {
      expect(getByText('O campo nome do paciente é obrigatório')).toBeDefined();
      expect(getByText('O campo data é obrigatório')).toBeDefined();
      expect(getByText('O campo hora é obrigatório')).toBeDefined();
    });
  });
});
