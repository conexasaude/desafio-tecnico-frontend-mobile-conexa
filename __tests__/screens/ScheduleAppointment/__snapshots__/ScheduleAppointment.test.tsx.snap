// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleAppointment should matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 32,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#7F7F7F",
            "fontFamily": "Nunito-SemiBold",
            "fontSize": 14,
            "marginBottom": 16,
          },
          undefined,
        ]
      }
    >
      Preencha os campos abaixo para agendar uma consulta com o paciente
    </Text>
    <View
      style={
        [
          {
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          [
            {},
            {},
          ]
        }
      >
        <View
          style={
            [
              {
                "borderColor": "#2E50D4",
                "borderRadius": 8,
                "borderWidth": 2,
                "height": 52,
                "marginTop": 16,
                "padding": 8,
              },
              undefined,
            ]
          }
          testID="input-container"
          textArea={false}
        >
          <Text
            style={
              [
                {
                  "backgroundColor": "#fff",
                  "color": "#2E50D4",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 14,
                  "left": 16,
                  "padding": 4,
                  "position": "absolute",
                  "top": -14,
                },
                undefined,
              ]
            }
          >
            Nome do paciente
          </Text>
          <View
            style={
              [
                {
                  "flex": 1,
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
          >
            <TextInput
              multiline={false}
              onChangeText={[Function]}
              placeholder="Ex: João da silva"
              placeholderTextColor="#B3B3B3"
              style={
                [
                  {
                    "color": "#333333",
                    "flex": 1,
                    "fontFamily": "Nunito-SemiBold",
                    "fontSize": 16,
                    "padding": 0,
                  },
                  undefined,
                ]
              }
              testID="input-patient"
              textAlignVertical="center"
              value=""
            />
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "flexDirection": "row",
              "gap": 16,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
              {},
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderColor": "#2E50D4",
                "borderRadius": 8,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "marginTop": 16,
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="date-picker-date"
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "#fff",
                    "color": "#2E50D4",
                    "fontFamily": "Nunito-Bold",
                    "fontSize": 14,
                    "left": 16,
                    "padding": 4,
                    "position": "absolute",
                    "top": -14,
                  },
                  undefined,
                ]
              }
            >
              Data
            </Text>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#333333",
                      "flex": 1,
                      "fontFamily": "Nunito-SemiBold",
                      "fontSize": 16,
                    },
                    undefined,
                  ]
                }
              >
                Selecione...
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#2E50D4",
                      "fontSize": 20,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰃭
              </Text>
            </View>
          </View>
        </View>
        <Modal
          hardwareAccelerated={false}
          visible={false}
        />
        <View
          style={
            [
              {
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
              },
              {},
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderColor": "#2E50D4",
                "borderRadius": 8,
                "borderWidth": 2,
                "height": 52,
                "justifyContent": "center",
                "marginTop": 16,
                "opacity": 1,
                "padding": 8,
              }
            }
            testID="date-picker-time"
          >
            <Text
              style={
                [
                  {
                    "backgroundColor": "#fff",
                    "color": "#2E50D4",
                    "fontFamily": "Nunito-Bold",
                    "fontSize": 14,
                    "left": 16,
                    "padding": 4,
                    "position": "absolute",
                    "top": -14,
                  },
                  undefined,
                ]
              }
            >
              Hora
            </Text>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#333333",
                      "flex": 1,
                      "fontFamily": "Nunito-SemiBold",
                      "fontSize": 16,
                    },
                    undefined,
                  ]
                }
              >
                Selecione...
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "#2E50D4",
                      "fontSize": 20,
                    },
                    undefined,
                    {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰃭
              </Text>
            </View>
          </View>
        </View>
        <Modal
          hardwareAccelerated={false}
          visible={false}
        />
      </View>
      <View
        style={
          [
            {},
            {},
          ]
        }
      >
        <View
          style={
            [
              {
                "borderColor": "#2E50D4",
                "borderRadius": 8,
                "borderWidth": 2,
                "height": 180,
                "marginTop": 16,
                "padding": 8,
              },
              undefined,
            ]
          }
          testID="input-container"
          textArea={true}
        >
          <Text
            style={
              [
                {
                  "backgroundColor": "#fff",
                  "color": "#2E50D4",
                  "fontFamily": "Nunito-Bold",
                  "fontSize": 14,
                  "left": 16,
                  "padding": 4,
                  "position": "absolute",
                  "top": -14,
                },
                undefined,
              ]
            }
          >
            Observação
          </Text>
          <View
            style={
              [
                {
                  "flex": 1,
                  "flexDirection": "row",
                },
                undefined,
              ]
            }
          >
            <TextInput
              multiline={true}
              numberOfLines={6}
              onChangeText={[Function]}
              placeholder="..."
              placeholderTextColor="#B3B3B3"
              style={
                [
                  {
                    "color": "#333333",
                    "flex": 1,
                    "fontFamily": "Nunito-SemiBold",
                    "fontSize": 16,
                    "padding": 0,
                  },
                  undefined,
                ]
              }
              testID="input-observation"
              textAlignVertical="top"
              value=""
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#2E50D4",
          "borderColor": "#2E50D4",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 0,
          "height": 52,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 8,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#fff",
              "fontFamily": "Nunito-Bold",
              "fontSize": 14,
              "textAlign": "center",
            },
            undefined,
          ]
        }
        variant="contained"
      >
        Agendar
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
